frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseRenameAction.java

```diff
@@ -96,18 +95,13 @@ private void rename(GitRepository gitRepository, BaseEnhancedGraphTable graphTab
       Runnable renameRunnable = () -> gitBrancher.renameBranch(branch.getName(), options.getName(),
           Collections.singletonList(gitRepository));
 
-      Runnable doInUIThreadWhenReady = new RenameBackgroundable(gitRepository,
+      new RenameBackgroundable(gitRepository,
```

Comment:
I'd even remove `RenameBackgroundable` for this PoC (just fire `gitBrancher.rename(...)` directly from the action code, as you did for a moment until today), to have less moving parts to analyze :async-parrot:

Reply:
ok